app/assets/javascripts/app/views/search_base_view.js

```diff
@@ -0,0 +1,141 @@
+app.views.SearchBase = app.views.Base.extend({
+  completeSetup: function(typeaheadElement){
+    this.typeaheadElement = $(typeaheadElement);
+    this.setupBloodhound();
+    this.setupTypeahead();
+    this.bindSelectionEvents();
+    this.resultsTofilter = [];
+  },
+
+  setupBloodhound: function() {
+    var self = this;
+    var bloodhoundConf = {
+      datumTokenizer: function(datum) {
+        var nameTokens = Bloodhound.tokenizers.nonword(datum.name);
+        var handleTokens = datum.handle ? Bloodhound.tokenizers.nonword(datum.name) : [];
+        return nameTokens.concat(handleTokens);
+      },
+      queryTokenizer: Bloodhound.tokenizers.whitespace,
+      prefetch: {
+        url: "/contacts.json",
+        transform: this.transformBloodhoundResponse,
+        cache: false
+      },
+      sufficient: 5
+    };
+
+    // The publisher does not define an additionnal source for searchin
+    // This prevents tests from failing when this additionnal source isn't set
+    if(this.searchFormAction !== undefined){
+      bloodhoundConf.remote = {
+        url: this.searchFormAction + ".json?q=%QUERY",
+        wildcard: "%QUERY",
+        transform: this.transformBloodhoundResponse
+      };
+    }
+
+    this.bloodhound = new Bloodhound(bloodhoundConf);
+
+    /**
+     * Custom searching function that let us filter contacts from prefetched Bloodhound results.
+     */
+    this.bloodhound.customSearch = function(query, sync, async){
+      var filterResults = function(datums){
+        return _.filter(datums, function(result){
+          if(result.handle){
+            return !_.contains(self.resultsTofilter, result.handle);
+          }
+        });
+      };
+
+      var _sync = function(datums){
+        var results = filterResults(datums);
+        sync(results);
+      };
+
+      self.bloodhound.search(query, _sync, async);
+    };
+  },
+
+  setupTypeahead: function() {
+    this.typeaheadElement.typeahead({
+          hint: false,
+          highlight: true,
+          minLength: 2
+        },
+        {
+          name: "search",
+          display: "name",
+          limit: 5,
+          source: this.bloodhound.customSearch,
```

Comment:
`source: this.searchFormAction === undefined ? this.bloodhound : this.bloodhound.customSearch` should work. If we need result filtering we can revert this later but calling `_.filter` although we don't need it sounds like it would make our code slower.